### PR TITLE
Reload button for No Connection Error

### PR DESCRIPTION
**Pull request by jrchudy**
Added reload button for when a No Connection Error occurs during the loading of the application.

### CODE REVIEW

**Comment by jrchudy** [comment]
@RFSH  the changes to show the reload button for all retriable error codes have been added to this PR.

File: common/modal.js [hunk 1]
```diff
@@ -60,6 +54,28 @@
...
+            }
+        }
+
+        var vm = this;
+        vm.params = params;
+        vm.displayDetails = false;
+        vm.linkText = messageMap.showErrDetails;
+        vm.showReloadBtn = false;
+        if (ERMrest && isRetryError(exception)) {
+            vm.showReloadBtn = !$rootScope.displayReady;
```
Comment (RFSH) [inline common/modal.js]: I think this means that we're only showing reload button on the requests that are being created before the page completely loads? Can you please add a comment as to why it's looking for `displayReady`?

Comment (RFSH) [reply common/modal.js]: I might be wrong, but it seems like recordset is not setting this variable.

Comment (jrchudy) [reply common/modal.js]: That's exactly the point, we are only showing this for the 4 types of retriable errors while the page is loading. Last week we discussed that it doesn't make sense to "retry" other requests that may fail after the data has loaded.

I made a change to table.js to set `$rootScope.displayReady` after the main entity request succeeds or fails.

Comment (RFSH) [reply common/modal.js]: My bad, I didn't see the `table.js` changes. Makes sense. I think we need to comment the reasoning in the code. It's not really obvious what the code is trying to do.

**Pull request merged**